extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java

```diff
@@ -63,6 +70,34 @@ public void scanFunctions(BeanArchiveIndexBuildItem beanArchiveIndexBuildItem,
             if (functionName != null && functionName.isEmpty())
                 functionName = null;
             functions.produce(new FunctionBuildItem(className, methodName, functionName));
+
+            String source = FunctionScannerBuildStep.class.getSimpleName() + " > " + method.declaringClass() + "[" + method
+                    + "]";
+
+            Type returnType = method.returnType();
+            if (returnType.kind() != Type.Kind.VOID) {
+                reflectiveHierarchy.produce(new ReflectiveHierarchyBuildItem.Builder()
```

Comment:
If the return type Uni<T> doesn't this register whole Uni<T>  instead of just T?
We do some processing related to this in `FunctionInvoker`.

If possible maybe registration of FunctionInvoker::inputType and FunctionInvoker::outputType should be sufficient.

Reply:
I'll check it out later on today or tomorrow morning

Reply:
PR updated to exclude Mutiny types (the generics types **do** get registered however).

We don't need the output of `FunctionInvoker::inputType` or `FunctionInvoker::outputType` as we already know the types that need to be registered at build time by just looking at the method signatures (same as RESTEasy)